lib/Buzz/Client/AbstractCurl.php

```diff
@@ -14,12 +14,14 @@
 abstract class AbstractCurl extends AbstractClient
 {
     protected $options = array();
+    protected static $headersSize = 0;
+    protected static $headers = array();
```

Comment:
Well, a global state is a very bad idea.

And btw, in case of MultiCurl, it is totally broken as you will share the *same* global state among all the requests being performed